### PR TITLE
Updating bin/tools/clear_package_properties.dart

### PR DESCRIPTION
**Pull request by isoos**
- added flag for `dry-run`
- added flag for `verbose` logging
- added stat collection

### CODE REVIEW

**Comment by isoos** [comment]
Btw. staging has the following properties to be cleared:
-  `Package.uploaderEmails`
- `PackageVersion.uploaderEmail`
-  `PackageVersionPubspec.namespace`
-  `PackageVersionPubspec.qualifiedPackage`
-  `PackageVersionInfo.namespace`
-  `PackageVersionInfo.qualifiedPackage`

File: app/bin/tools/clear_package_properties.dart [hunk 1]
```diff
@@ -27,68 +35,91 @@ Future main(List<String> args) async {
...
-      tx.queueMutations(inserts: [p]);
-      await tx.commit();
-    });
+  if (package.additionalProperties.isNotEmpty) {
+    await _clearAdditionalProperties(package);
+  }
+}
+
+Future _clearAdditionalProperties<T extends ExpandoModel>(T model) async {
+  final props = model.additionalProperties.keys.toList();
```
Comment (jonasfj) [inline app/bin/tools/clear_package_properties.dart]: Can't we move `if (model.additionalProperties.isEmpty) { return; }` in here..?

File: app/bin/tools/clear_package_properties.dart [hunk 2]
```diff
@@ -27,68 +35,91 @@ Future main(List<String> args) async {
...
-  bool packageEntityNeedsUpdate = true,
-}) async {
-  print('Processing package: $package');
-  final pkgKey = dbService.emptyKey.append(Package, id: package);
+Future _processPackage(Package package) async {
+  print('Processing package: ${package.name}');
 
-  final versionQuery = dbService.query<PackageVersion>(ancestorKey: pkgKey);
+  final versionQuery =
+      dbService.query<PackageVersion>(ancestorKey: package.key);
```
Comment (jonasfj) [inline app/bin/tools/clear_package_properties.dart]: This won't work, as you can only do one transaction per second on keys with the same parent... I would assume this causes problems at some point...

Given the size of these, I'm also not sure it's wise to do all of these in one transaction.. but that might be an option.
We would likely need to merge such a transaction with `_clearAdditionalProperties(package)`.

Comment (isoos) [reply app/bin/tools/clear_package_properties.dart]: I've tried it on staging, and it works. It does throttle the operations one second at a time, but it does not fail. We had similar processing previously in other scripts, and it was the same behaviour.

Comment (jonasfj) [reply app/bin/tools/clear_package_properties.dart]: Okay, then let's give it a shot :)

**Pull request merged**